### ISSUE TITLE
TypeScript code completion

### ISSUE THREAD

**Issue by stoskov** [ternjs/tern#317]
Hi Marijn,

It is rather feature request than issue, but I think a lot of people are interested in the answer.

As far as I know TernJS could currently only analyse JavaScript code and could not "understand" TypeScript, right? 
Is there a plan for TernJS to analyse TypeScript and support code completion in TypeScript files in near future? 
As its community grows every day it will be awesome to have it built in. 

Thank you!

**Comment by RReverser** [comment]
Hi all. I know this is an old issue, but I was also playing with TypeScript API for a long time, and have to add that they finally stabilized and improved it a lot starting from TypeScript 1.5. So it should be possible to fix this issue as soon as someone willing to spend a bit of time to adapt to the new (non-breaking) API.

**Comment by dannysu** [comment]
If anyone wants to collaborate, I've migrated `from_ts` and related code to [dannysu/dts2tern](https://github.com/dannysu/dts2tern). It's been updated with the new way of accessing the TypeScript parser and AST.

Currently running `./bin/from_ts node_modules/definitely-typed/exit/exit.d.ts` will produce:
```
{
  "exit": {
    "exit": "fn(code: number)"
  },
  "!name": "exit.d.ts"
}
```

It's a work-in-progress. I have no idea if the produced tern file is correct or not. With the time I spent on it today so far, I only got to it kinda running.
